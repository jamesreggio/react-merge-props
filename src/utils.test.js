import {Component} from 'react';

import {
  invariant,
  getDisplayName,
} from './utils';

describe(`invariant`, () => {
  it(`throws when the condition is falsey`, () => {
    expect(() => invariant(null, 'Error message')).toThrowErrorMatchingSnapshot();
  });

  it(`doesn't throw when the condition is truthy`, () => {
    expect(() => invariant({}, 'Error message')).not.toThrow();
  });
});

describe(`getDisplayName`, () => {
  it(`handles React class components`, () => {
    class MyComponent extends Component { }
    expect(getDisplayName(MyComponent)).toBe('MyComponent');
  });

  it(`handles React stateless components`, () => {
    const MyComponent = () => {};
    expect(getDisplayName(MyComponent)).toBe('MyComponent');
  });

  it(`handles unknown components`, () => {
    const UnknownComponent = {};
    expect(getDisplayName(UnknownComponent)).toBe('Component');
  });

  it(`respects overridden component display names`, () => {
    class MyComponent extends Component { }
    MyComponent.displayName = 'OtherComponent';
    expect(getDisplayName(MyComponent)).toBe('OtherComponent');
  });
});
