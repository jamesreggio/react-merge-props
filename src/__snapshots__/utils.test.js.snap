// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`invariant throws when the condition is falsey 1`] = `"Error message"`;
